galley/pkg/config/analysis/analyzers/auth/mtls.go

```diff
@@ -0,0 +1,185 @@
+package auth
+
+import (
+	"istio.io/api/authentication/v1alpha1"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+
+	"istio.io/api/networking/v1alpha3"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/auth/mtls"
+	"istio.io/istio/galley/pkg/config/meta/metadata"
+	"istio.io/istio/galley/pkg/config/meta/schema/collection"
+	"istio.io/istio/galley/pkg/config/resource"
+)
+
+// MTLSAnalyzer checks for misconfigurations of MTLS policy. More specifically,
+// it detects situations where a DestinationRule's MTLS usage is in conflict
+// with mTLS specified by a policy.
```

Comment:
This should also take into account auto-mtls. The omission of a DR w/mTLS is not necessarily an error if Pilot is automatically configuring client TLS settings based on the server side auth policy. 

cc @incfly to verify correct of the auto-mtls logic.

Reply:
@ayj you’re correct AFAIK but the converse is true too i.e. if you have a DR with auto-mTLS on, rules from DR takes precedence for the clients. To summarize missing DR in AutoMTLS or permissive should work but a DR specified shouldn’t conflict in any case is my understanding.

Reply:
This is something I was hoping to avoid since it adds a lot of complexity, and is turned off by default. The code is already pretty complicated.

That said, to move forward, how about I consume MeshConfig and disable the entire validator if automtls is enabled? A followup CL can add support for reasoning about automtls behavior.

Reply:
We make DestinationRule ISTIO_MUTUAL take higher priority for the sake of backward compatibility. But as far as I know, going forward, we are moving to not having today's DestinationRule at all, deprecate ISTIO_MUTUAL specifically.

This tool is used for the CI/CD pipeline, I think it's important to encourage customers to move to our future direction as well. So it's fine to have a incremental PR first, but I really think we have to cover auto mTLS in a released version of `istioctl`.

Maybe we can have an issue to track that?

Reply:
AFAICT, auto-mtls is enabled by default in master. 

@incfly, what types of incompatibilities exist when auto-mtls=true? My understanding was that Pilot automatically "does the right thing" and prevents the types of errors this analysis (and authn tls-check) looks for.

Reply:
I think you're still able to cause conflicts via usage of ISTIO_MUTUAL.

Auto-mtls is enabled in master; I do think we should backport this to 1.4 however as it'd be pretty easy to include in the next patch release. I also think there will be some period of time when people are adopting automtls where this could be potentially useful.

As an addendum, I spoke with @incfly more and we floated the idea of an analyzer that warns about DRs that use ISTIO_MUTUAL on an automtls enabled mesh. It probably wont live in this code but it's worth exploring.